anton.baraiev/task_5/src/main/java/com/hillel/BankAccountServlet.java

```diff
@@ -0,0 +1,133 @@
+            long id = createBankAccountRequest.getId();
+            if (accounts.containsKey(id)) {
+                BankAccount bankAccount = new BankAccount(id, createBankAccountRequest.getAmount(), createBankAccountRequest.getUser());
+                accounts.put(id, bankAccount);
+                message = String.format("Bank account id = %d info was updated", id);
+                status = new CrudEventStatus(true, message);
+                log.info(message);
+            }else {
+                message = String.format("There is no bank account w/ given id = %d", id);
+                status = new CrudEventStatus(false, message);
+                log.warn(message);
+                resp.setStatus(404);
+            }
+        }catch (Exception e) {
+            message = "Error, while trying to update account info";
+            status = new CrudEventStatus(false, message);
+            resp.setStatus(500);
+        }
+        resp.getWriter().println(status.getDescription());
+        resp.getWriter().flush();
+    }
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+        resp.setContentType("application/json");
+        Gson gson = new Gson();
+        String message;
+        CrudEventStatus status;
+        if (req.getParameter("id") != null) {
+            try {
+                long accountId = Long.parseLong(req.getParameter("id"));
+                if (accounts.containsKey(accountId)) {
+                    message = gson.toJson(accounts.get(accountId));
+                    status = new CrudEventStatus(true, message);
+                }else {
+                    message = "There is no account with such id";
+                    status = new CrudEventStatus(false, message);
+                    resp.setStatus(404);
+                }
+            }catch (Exception e) {
+                message = "Error while trying to get account info";
+                status = new CrudEventStatus(false, message);
+                resp.setStatus(500);
+            }
+        }else {
+            message = "Bad request";
+            status = new CrudEventStatus(false, message);
+            resp.setStatus(500);
+        }
+        resp.getWriter().println(status.getDescription());
+        resp.getWriter().flush();
+    }
+
+    @Override
+    protected void doDelete(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+        resp.setContentType("application/json");
+        String message;
+        CrudEventStatus status;
+        try{
+            if(req.getParameter("id") != null) {
+                long id = Long.parseLong(req.getParameter("id"));
+                accounts.remove(id);
+                message = String.format("Account id = %d was successfully removed", id);
+                status = new CrudEventStatus(true, message);
+                log.info(message);
+            } else {
+                message = "There is no account with given id";
```

Comment:
If id is null - message should be another. For example "Missed required request parameter 'id'"